app/etl/ga/concerns/transform_path.rb

```diff
@@ -3,12 +3,29 @@
 module GA::Concerns::TransformPath
   extend ActiveSupport::Concern
 
-  def remove_invalid_prefix
+  def format_events_with_invalid_prefix
     events_with_prefix = Events::GA.where("page_path ~ '^\/https:\/\/www.gov.uk'")
     log(process: :ga, message: "Transforming #{events_with_prefix.count} events with page_path starting https://gov.uk")
     events_with_prefix.find_each do |event|
-      page_path_without_prefix = event.page_path.remove '/https://www.gov.uk'
-      event.update_attributes(page_path: page_path_without_prefix)
+      transformed_attributes = transform_event_attributes(event)
+      event.update_attributes(transformed_attributes)
     end
   end
+
+private
+
+  def transform_event_attributes(event)
+    sanitised_page_path = event.page_path.remove '/https://www.gov.uk'
+    duplicate_event = Events::GA.find_by(page_path: sanitised_page_path)
```

Comment:
What about adding a constrain so we can't have duplicates for the same process name and page_path? This constraint would have raise the problem earlier.

Reply:
We have agreed to do it in a separate PR.